test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/MiddleWareTests.cs

```diff
@@ -336,5 +336,35 @@ public class MiddlewareTests
 
             Assert.Equal(response.Headers.Location.OriginalString, "/foo");
         }
+
+        [Theory]
+        [InlineData("IsFile")]
+        [InlineData("isfile")]
+        [InlineData("IsDirectory")]
+        [InlineData("isdirectory")]
+        public async Task VerifyIsFileAndIsDirectoryParsing(string matchType)
```

Comment:
This isn't really testing that the parsing is working, before an invalid value would just make `matchType = Pattern` and now the same thing will still happen. We need to verify that `IsFile` and `IsDirectory` are being used